include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -3142,6 +3150,163 @@ class FEFaceEvaluation : public FEEvaluationAccess<dim,
    * points is inaccurate and this value must be used instead.
    */
   const unsigned int n_q_points;
+
+
+private:
+  /**
+   * Return face number of each face of the current face batch.
+   */
+  std::array<unsigned int, VectorizedArrayType::size()>
+  compute_face_no_data()
+  {
+    std::array<unsigned int, VectorizedArrayType::size()> face_no_data;
```

Comment:
I think we regularly don't implement functions inline in user-visible classes (rather than internal ones), so this would need to be moved to the implementations below.